rpcs3/rpcs3qt/infinity_dialog.cpp

```diff
@@ -0,0 +1,831 @@
+			combo_figlist->addItem(QString::fromStdString(entry.second.second), QVariant(qnum));
+			filterlist << entry.second.second.c_str();
+			if (first_entry == 0)
+			{
+				first_entry = figure;
+			}
+		}
+	}
+
+	combo_figlist->addItem(tr("--Unknown--"), QVariant(0xFFFFFFFF));
+	combo_figlist->setEditable(true);
+	combo_figlist->setInsertPolicy(QComboBox::NoInsert);
+
+	QCompleter* co_compl = new QCompleter(filterlist, this);
+	co_compl->setCaseSensitivity(Qt::CaseInsensitive);
+	co_compl->setCompletionMode(QCompleter::PopupCompletion);
+	co_compl->setFilterMode(Qt::MatchContains);
+	combo_figlist->setCompleter(co_compl);
+
+	vbox_panel->addWidget(combo_figlist);
+
+	QFrame* line = new QFrame();
+	line->setFrameShape(QFrame::HLine);
+	line->setFrameShadow(QFrame::Sunken);
+	vbox_panel->addWidget(line);
+
+	QHBoxLayout* hbox_number = new QHBoxLayout();
+	QLabel* label_number = new QLabel(tr("Figure Number:"));
+	QLineEdit* edit_number = new QLineEdit(QString::fromStdString(std::to_string(first_entry)));
+	QLabel* label_series = new QLabel(tr("Series:"));
+	QLineEdit* edit_series = new QLineEdit("1");
+	QRegularExpressionValidator* rxv = new QRegularExpressionValidator(QRegularExpression("\\d*"), this);
+	QIntValidator* valid_series = new QIntValidator(1, 3, this);
+	edit_number->setValidator(rxv);
+	edit_series->setValidator(valid_series);
+	hbox_number->addWidget(label_number);
+	hbox_number->addWidget(edit_number);
+	hbox_number->addWidget(label_series);
+	hbox_number->addWidget(edit_series);
+	vbox_panel->addLayout(hbox_number);
+
+	QHBoxLayout* hbox_buttons = new QHBoxLayout();
+	QPushButton* btn_create = new QPushButton(tr("Create"), this);
+	QPushButton* btn_cancel = new QPushButton(tr("Cancel"), this);
+	hbox_buttons->addStretch();
+	hbox_buttons->addWidget(btn_create);
+	hbox_buttons->addWidget(btn_cancel);
+	vbox_panel->addLayout(hbox_buttons);
+
+	setLayout(vbox_panel);
+
+	connect(combo_figlist, QOverload<int>::of(&QComboBox::currentIndexChanged), [=](int index)
+		{
+			const u32 fig_info = combo_figlist->itemData(index).toUInt();
+			if (fig_info != 0xFFFFFFFF)
+			{
+				const u32 fig_num = fig_info >> 8;
+				const u8 series = fig_info & 0xFF;
+
+				edit_number->setText(QString::number(fig_num));
+				edit_series->setText(QString::number(series));
+			}
+		});
+
+	connect(btn_create, &QAbstractButton::clicked, this, [=, this]()
+		{
+			bool ok_num = false, ok_series = false;
+			const u32 fig_num = edit_number->text().toULong(&ok_num);
+			if (!ok_num)
+			{
+				QMessageBox::warning(this, tr("Error converting value"), tr("Figure number entered is invalid!"), QMessageBox::Ok);
+				return;
+			}
+			const u8 series = edit_series->text().toUShort(&ok_series);
+			if (!ok_series || series > 3 || series < 1)
+			{
+				QMessageBox::warning(this, tr("Error converting value"), tr("Series number entered is invalid!"), QMessageBox::Ok);
+				return;
+			}
+			QString predef_name = s_last_figure_path;
```

Comment:
Redundant assignment
```suggestion
			QString predef_name;
```